Tella/Data/Networking/Repositories/UwaziServerRepositories.swift

```diff
@@ -139,6 +139,23 @@ class UwaziServerRepository: WebRepository {
             }
             .eraseToAnyPublisher()
         }
+    
+    func getRelationshipEntities(serverURL: String, cookie: String, templatesIds: [String?]) -> AnyPublisher<[UwaziRelationshipList], APIError> {
+        let apiResponse: APIResponse<UwaziRelationshipDTO> = getAPIResponse(endpoint: API.getRelationshipEntities(serverURL:serverURL, cookie:cookie))
+
+        return apiResponse
+            .compactMap{ $0.0 }
+            .map { dto in
+                let shouldFetchAllTemplates = templatesIds.contains { $0?.isEmpty ?? false }
+                return dto.rows
+                    .filter{ $0.type == "template" }
+                    .filter { relationship in
+                        shouldFetchAllTemplates || templatesIds.contains(where: { $0 == relationship.id })
+                    }
```

Comment:
We can Combine the filtering operations for better readability ->  
`relationship in shouldFetchAllTemplates || templateIds.contains(relationship.id)`